internal/uvm/scsi.go

```diff
@@ -323,11 +283,40 @@ func (uvm *UtilityVM) addSCSIActual(ctx context.Context, hostPath, uvmPath, atta
 	}
 
 	if err := uvm.modify(ctx, SCSIModification); err != nil {
-		return nil, fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
+		return nil, fmt.Errorf("failed to modify UVM with new SCSI mount: %s", err)
 	}
 	return sm, nil
 }
 
+// allocateSCSIMount grants vm access to hostpath and increments the ref count of an existing scsi
+// device or allocates a new one if not already present.
+// Returns the resulting *SCSIMount, a bool indicating if the scsi device was already present,
+// and error if any.
+func (uvm *UtilityVM) allocateSCSIMount(ctx context.Context, hostPath, uvmPath string, vmAccess VMAccessType) (*SCSIMount, bool, error) {
+	// Ensure the utility VM has access
+	err := uvm.grantAccess(ctx, hostPath, vmAccess)
+	if err != nil {
+		return nil, false, errors.Wrapf(err, "failed to grant VM access for SCSI mount")
+	}
+	// We must hold the lock throughout the lookup (findSCSIAttachment) until
+	// after the possible allocation (allocateSCSISlot) has been completed to ensure
+	// there isn't a race condition for it being attached by another thread between
+	// these two operations. All failure paths between these two must release
```

Comment:
We don't really need this comment about failure paths anymore, since we just defer the unlock.

Reply:
I'm fine removing the last part about all failure paths, but I'd like to keep the rest so it's clear why a lock is being taken :)

Reply:
Sounds good, I was only referring to the part about failure paths.